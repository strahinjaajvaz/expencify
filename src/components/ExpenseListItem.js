import React from "react";
import { Link } from "react-router-dom";

export default ({ description, amount, createdAt, id }) => {
  return (
    <div>
      <h3>
        <Link to={`/edit/${id}`}>Description: {description}</Link>
      </h3>
      <p>amount: {amount}</p>
      <p>createdAt: {createdAt}</p>
    </div>
  );
};
