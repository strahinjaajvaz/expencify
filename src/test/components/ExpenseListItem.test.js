import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ExpenseListItem from "../../components/ExpenseListItem";

test("should redner ExpenseListItem with expenses", () => {
  const wrapper = shallow(
    <ExpenseListItem
      expense={{ description: "Water Bill", amount: 1200, createdAt: 0, id: 1 }}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
