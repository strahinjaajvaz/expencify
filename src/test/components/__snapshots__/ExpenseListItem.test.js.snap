// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redner ExpenseListItem with expenses 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/undefined"
    >
      Description: 
    </Link>
  </h3>
  <p>
    amount: 
  </p>
  <p>
    createdAt: 
  </p>
</div>
`;
